arroyo/processing/strategies/dead_letter_queue/policies/abstract.py

```diff
@@ -1,18 +1,37 @@
 from abc import ABC, abstractmethod
-from typing import Any, Sequence
+from datetime import datetime
+from typing import Optional, Sequence, Union
+
+Serializable = Union[str, bytes]
 
 
 class InvalidMessages(Exception):
     """
     An exception to be thrown to pass bad messages to the DLQ
     so they are handled correctly.
+
+    If the original topic the message(s) were produced to is known,
+    that should be passed along via this exception.
     """
 
-    def __init__(self, messages: Sequence[Any]):
+    def __init__(
+        self,
+        messages: Sequence[Serializable],
+        reason: Optional[str] = None,
+        original_topic: Optional[str] = None,
+    ):
         self.messages = messages
+        self.reason = reason or "unknown"
+        self.topic = original_topic or "unknown"
+        self.timestamp = str(datetime.now())
```

Comment:
Please do not use strings for time stamps.
Either an epoch timestamp (float) or a DateTime object.

Using strings means you can store whatever you want in the timestamp and not just a valid timestamp.

Reply:
A `datetime` object isn't JSON serializable, that timestamp is really only meant to show when the exception itself was raised and it isn't a parameter of the exception itself. I'll change this and stringify at json serialize time.

Reply:
I guess the time at which the exception was generated doesn't really matter for now, I'll remove it entirely.

Reply:
If you need to json serialize `datetime` objects you can pass a function for that type 
https://github.com/getsentry/snuba/blob/master/snuba/clickhouse/http.py#L38-L47